src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -171,7 +171,9 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 		}
 
 		if (environmentVariableCollections) {
-			this._extEnvironmentVariableCollection = new MergedEnvironmentVariableCollection(environmentVariableCollections);
+			const activeWorkspaceRootUri = this._historyService.getLastActiveWorkspaceRoot();
+			const lastActiveWorkspace = activeWorkspaceRootUri ? withNullAsUndefined(this._workspaceContextService.getWorkspaceFolder(activeWorkspaceRootUri)) : undefined;
```

Comment:
Is this the way to get the workspace folder?

Reply:
Yes this looks right, since the process workspaces seems to be used 3 spots now, let's move it to a private member. Without doing this I think there might be a race condition if the workspace changes between `ctor` and `createProcess`.

Reply:
So I create a private property `activeWorkspace` which is calculated in the constructor and never changes?